Hackathon_Converter/MainForm.cs

```diff
@@ -51,63 +55,495 @@ private void OnAddFilesButtonClick(object sender, EventArgs e)
-            htmlStream.Write(bytes, 0, bytes.Length);
-            htmlStream.Close();
 
-            var htmlFilename = saveFileDialog.FileName;
-            var dirname = Path.GetDirectoryName(htmlFilename);
-            var cssFilename = Path.Combine(dirname, "style.css");
-            if (File.Exists(cssFilename) == false)
+            
+            if (fileDataGrid.Rows.Count == 1)
+                numRemove = 0;
+
+            btn_Remove.Text = $"Delete ({numRemove}) Files";
+
+
+        }
+
+        private void Btn_NewProj_Click(object sender, EventArgs e)
+        {
+            Show_Home_Page();
+            fileDataGrid.Rows.Clear();
+        }
+
+        private void Btn_OpenFileLocation_Click(object sender, EventArgs e)
+        {
+            if (filePathConversion == null)
+                Process.Start("explorer.exe");
+            else
+                Process.Start(filePathConversion);
+        }
+        private void fileDataGrid_CellStateChanged(object sender, DataGridViewCellStateChangedEventArgs e)
+        {
+            
+            DataGridViewElementStates state = e.StateChanged;
+            DataGridViewSelectedCellCollection SelectedFiles = fileDataGrid.SelectedCells;
+
+
+            int numFilesRemove = SelectedFiles.Count;
+            if (SelectedFiles.Contains(fileDataGrid[0, fileDataGrid.RowCount-1]))
+                numFilesRemove--;
+
+            //if (SelectedFiles.Contains(fileDataGrid.Rows[fileDataGrid.RowCount - 1].Cells[1]))
+            //    numFilesRemove--;
+            btn_Remove.Text = $"Delete ({numFilesRemove}) Files";
+        }
+
+        private void Btn_Format_Click(object sender, EventArgs e)
+        {
+            Show_Format_Page();
+        }
+
+        private void Btn_Spaced_Click(object sender, EventArgs e)
+        {
+            // Pseudo Radio Button Styling
+            isSingleSpaced = !isSingleSpaced;
+            if (isSingleSpaced)
             {
-                File.Copy("style.css", cssFilename);
+                this.Btn_SingleSpaced.BackColor = System.Drawing.Color.White;
+                this.Btn_SingleSpaced.FlatAppearance.BorderColor = System.Drawing.Color.FromArgb(((int)(((byte)(0)))), ((int)(((byte)(68)))), ((int)(((byte)(214)))));
+                this.Btn_SingleSpaced.ForeColor = System.Drawing.Color.FromArgb(((int)(((byte)(0)))), ((int)(((byte)(68)))), ((int)(((byte)(214)))));
+
+                // Grayed Out Button
+                this.Btn_DoubleSpaced.BackColor = System.Drawing.Color.FromArgb(((int)(((byte)(215)))), ((int)(((byte)(218)))), ((int)(((byte)(224)))));
+                this.Btn_DoubleSpaced.FlatAppearance.BorderColor = System.Drawing.Color.FromArgb(((int)(((byte)(118)))), ((int)(((byte)(118)))), ((int)(((byte)(118)))));
+                this.Btn_DoubleSpaced.ForeColor = System.Drawing.Color.FromArgb(((int)(((byte)(118)))), ((int)(((byte)(118)))), ((int)(((byte)(118)))));
+
+                configHTML.divClasses[0] = LineSpacingClasses[1.0];

```

Comment:
What I would do with this is just store the value in a private var and then set the div classes right before you actually call the renderer. Arbitrary indexes in an array isn't that readable or maintainable.

Reply:
Fixed all cases and put them into a function that gets called right before conversion

Reply:
Not all cases. I found some more 😜

Reply:
Should be the last of them!!